src/battle_script_commands.c

```diff
@@ -5725,6 +5726,22 @@ static void Cmd_moveend(void)
             else
                 gBattleStruct->lastMoveFailed &= ~(gBitTable[gBattlerAttacker]);
 
+            // Set ShellTrap to activate after the attacker's turn if target was hit by a physical move.
+            if (gBattleMoves[gBattleMons[gBattlerTarget].moves[gBattleStruct->chosenMovePositions[gBattlerTarget]]].effect == EFFECT_SHELL_TRAP
```

Comment:
This is ridiculously narrow, but in a double-battle where I KO one of the foes and replace the move I chose with Shell Trap, could it then activate on the same turn despite not having been chosen?

Reply:
You're totally right, I think this is exactly what would happen. I think it could be easily fixed tho

Reply:
Fixed